src/client/connect/mod.rs

```diff
@@ -24,26 +24,49 @@ use crate::common::{Future, Unpin};
 /// A connector receives a [`Destination`](Destination) describing how a
 /// connection should be estabilished, and returns a `Future` of the
 /// ready connection.
-pub trait Connect: Send + Sync {
+///
+/// # Trait Alias
+///
+/// This is really just an *alias* for the `tower::Service` trait, with
+/// additional bounds set for convenience *inside* hyper. You don't actually
+/// implement this trait, but `tower::Service<Destination>` instead.
+// The `Sized` bound is to prevent creating `dyn Connect`, since they cannot
+// fit the `Connect` bounds because of the blanket impl for `Service`.
+pub trait Connect: sealed::Sealed + Sized {
     /// The connected IO Stream.
-    type Transport: AsyncRead + AsyncWrite + Unpin + Send + 'static;
+    type Transport: AsyncRead + AsyncWrite;
     /// An error occured when trying to connect.
     type Error: Into<Box<dyn StdError + Send + Sync>>;
     /// A Future that will resolve to the connected Transport.
-    type Future: Future<Output=Result<(Self::Transport, Connected), Self::Error>> + Unpin + Send;
-    /// Connect to a destination.
-    fn connect(&self, dst: Destination) -> Self::Future;
+    type Future: Future<Output=Result<(Self::Transport, Connected), Self::Error>>;
```

Comment:
is this a good time to move to http connection?

Reply:
Yea, I started working on that, but ran into some questions, so figured I'd just update this much and continue *that* in a new PR.

Reply:
sounds like a good plan 👍